core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -332,4 +339,26 @@ public Path tmpFile() {
     public static FileStore getFileStore(Path path) throws IOException {
         return ESFileStore.getMatchingFileStore(path, fileStores);
     }
+
+    /**
+     * assserts that the two environments are equivalent for all things the environment cares about (i.e., all but the setting
+     * object which may contain different setting)
+     */
+    public static void assertEquivalent(Environment actual, Environment expected) {
+        assertEquals(actual.dataWithClusterFiles(), expected.dataWithClusterFiles(), "dataWithClusterFiles");
+        assertEquals(actual.repoFiles(), expected.repoFiles(), "repoFiles");
+        assertEquals(actual.configFile(), expected.configFile(), "configFile");
+        assertEquals(actual.scriptsFile(), expected.scriptsFile(), "scriptsFile");
+        assertEquals(actual.pluginsFile(), expected.pluginsFile(), "pluginsFile");
+        assertEquals(actual.binFile(), expected.binFile(), "binFile");
+        assertEquals(actual.libFile(), expected.libFile(), "libFile");
+        assertEquals(actual.modulesFile(), expected.modulesFile(), "modulesFile");
+        assertEquals(actual.logsFile(), expected.logsFile(), "logsFile");
+        assertEquals(actual.pidFile(), expected.pidFile(), "pidFile");
+        assertEquals(actual.tmpFile(), expected.tmpFile(), "tmpFile");
+    }
+
+    private static void assertEquals(Object actual, Object expected, String name) {
+        assert Objects.deepEquals(actual, expected) : "actual " + name + " [" + actual + "] is different than [ " + expected + "]";
```

Comment:
Are we sure that the array-valued attributes (e.g., `dataWithClusterFiles`) will always be in the same order?